### PR TITLE
Add fold support on device

### PR DESCRIPTION
**Pull request by yugaoTT**
### Ticket
[Link to Github Issue](https://github.com/tenstorrent/tt-metal/issues/10288)

### Problem description
need to have fold works on device for WH

### What's changed
impl fold as a chain of transpose, reshape, pad, and slice
add RM support for transposes hw/hc, reshape, pad

### Checklist
- [ ] Post commit CI passes https://github.com/tenstorrent/tt-metal/actions/runs/10116391469
- [ ] device perf pass (fail same as main)
- [ ] model perf pass https://github.com/tenstorrent/tt-metal/actions/runs/10148084320

### CODE REVIEW

**Comment by yugaoTT** [comment]
fyi @tt-aho @mywoodstock

File: tests/tt_eager/python_api_testing/unit_testing/misc/test_fold_op.py [hunk 1]
```diff
@@ -19,6 +36,139 @@ def fold_torch(input_tensor, stride_h, stride_w):
...
+    )
+    # unpad params
+    n, c, h, w = activation_pyt_padded.shape
+    target_h = h // stride_h
+    target_w = w // stride_w
+    # pad to 256, 256
+    n, c, h, w = activation_pyt_padded.shape
+    pad_h = 256 - h
+    pad_w = 256 - w
+    activation_pyt_padded = torch.nn.functional.pad(activation_pyt_padded, (0, pad_w, 0, pad_h))
```
Comment (mywoodstock) [inline tests/tt_eager/python_api_testing/unit_testing/misc/test_fold_op.py:53]: shouldn't the padding be `pad_w / 2` and `pad_h / 2` on both sides?

**Review by arakhmati — APPROVED** [review]
